aws_lambda_builders/workflows/go_modules/workflow.py

```diff
@@ -0,0 +1,51 @@
+"""
+Go Modules Workflow
+"""
+from aws_lambda_builders.workflow import BaseWorkflow, Capability
+
+from .actions import GoModulesBuildAction
+from .builder import GoModulesBuilder
+from .validator import GoRuntimeValidator
+from .utils import OSUtils
+
+
+class GoModulesWorkflow(BaseWorkflow):
+
+    NAME = "GoModulesBuilder"
+
+    CAPABILITY = Capability(language="go",
+                            dependency_manager="modules",
+                            application_framework=None)
+
+    def __init__(self,
+                 source_dir,
+                 artifacts_dir,
+                 scratch_dir,
+                 manifest_path,
+                 runtime=None,
+                 osutils=None,
+                 **kwargs):
+
+        super(GoModulesWorkflow, self).__init__(
+            source_dir,
+            artifacts_dir,
+            scratch_dir,
+            manifest_path,
+            runtime=runtime,
+            **kwargs)
+
+        if osutils is None:
+            osutils = OSUtils()
+
+        options = kwargs.get("options") or {}
```

Comment:
Not a huge deal but you can do `kwargs.get("options", {})`.

Reply:
I did it this way because if `options` is passed in as a kwarg set to `None`, then this would throw an exception. Using an `or` ensures we get a dict.

Reply:
We need an explicit None check then. if `kwargs` is `None` then this will fail with an `AttributeError: 'NoneType' object has no attribute 'get'`. Can you add a None check here?

I did this in a 3.7 interpreter:
```
>>> kwargs = None
>>> kwargs.get("something") or {}
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: 'NoneType' object has no attribute 'get'
```

Reply:
Nevermind. That will not happen. What you are talking about is the value of `kwargs = {'options': None}` not `kwargs = None`. I get it now (apparently I need another coffee :))

Reply:
Right, sorry my wording wasn't the best :)